import-scripts/annotation_suite/standardize_mutation_data.py

```diff
@@ -0,0 +1,1391 @@
+            header = list(map(str.strip, line.split("\t")))
+            break
+    return header
+
+def process_datum(value):
+    """
+        Returns a cleaned up data value.
+        A null or 'empty' value is always returned as an empty string.
+    """
+    try:
+        vfixed = value.strip()
+    except AttributeError:
+        vfixed = ""
+    if vfixed in ["", "", "N/A", None, ".", "?"]:
+        return ""
+    return vfixed.replace('"',"")
+
+def resolve_tumor_seq_allele(data, ref_allele):
+    """ Resolve the tumor seq allele. """
+    tum_seq_allele1 = ""
+    tum_seq_allele2 = ""
+
+    if MUTATED_TO_ALLELE_COLUMN in data.keys():
+        return data[MUTATED_TO_ALLELE_COLUMN]
+
+    for i,column in enumerate(TUMOR_SEQ_ALLELE1_COLUMNS):
+        if column in data.keys():
+            tum_seq_allele1 = process_datum(data[column])
+            tum_seq_allele2 = process_datum(data[TUMOR_SEQ_ALLELE2_COLUMNS[i]])
+
+            # if at least one is not empty then exit for-loop
+            if tum_seq_allele1 != "" or tum_seq_allele2 != "":
+                break
+
+    # if temp tum seq alleles are not equal then set tum_seq_allele that doesn't equal ref_allele
+    # otherwise there is probably something wrong with the data
+    if tum_seq_allele1 == "" and tum_seq_allele2 == "":
+        return ""
+    if tum_seq_allele1 == ref_allele:
+        return tum_seq_allele2
+    else:
+        return tum_seq_allele1
+
+
+def print_warning(message):
+    print(message, ERROR_FILE)
+
+def resolve_variant_type(data, ref_allele, tum_seq_allele):
+    """ Resolve variant type.  """
+    for column in VARIANT_TYPE_COLUMNS:
+        if column in data.keys():
+            variant_type = process_datum(data[column].upper())
+            break
+    if variant_type == "1":
+        return "SNP"
+    elif variant_type == "2":
+        return "INS"
+    elif variant_type == "3":
+        return "DEL"
+
+    # if variant type is empty or not valid then try
+    # resolving it based on ref allele and tum seq allele values
+    if not variant_type in VALID_VARIANT_TYPES:
+        if len(ref_allele) > len(tum_seq_allele):
+            variant_type = "DEL"
+        elif len(ref_allele) < len(tum_seq_allele):
+            variant_type = "INS"
+        elif len(ref_allele) == len(tum_seq_allele):
+            if len(ref_allele) == 1:
+                variant_type = "SNP"
+            elif len(ref_allele) == 2:
+                variant_type = "DNP"
+            elif len(ref_allele) == 3:
+                variant_type = "TNP"
+            else:
+                variant_type = "ONP"
+
+    if variant_type == "":
+        message = "Could not salvage variant type from alleles [ ref allele = %s , tumor allele = %s ] " % (ref_allele, tum_seq_allele)
+        print_warning(message)
+
+    # if variant type still not in VALID_VARIANT_TYPES then must be SUB/SNV
+    if variant_type == "SNV":
+        variant_type = "SNP"
+    elif variant_type.upper() == "SUB":
```

Comment:
If SUB is a synonym for "Substitution" then I wonder if this used to be a code representing an indel. Trying to understand the ONP definition, the MAF_Format document above implies (in my opinion) that the reference allele length will match the variant allele length. It says "ONP  is analogous to TNP but for consecutive runs of four or more". So if SNP is 1 for 1, and DNP is 2 for 2, and TNP is 3 for 3, then ONP (by analogy) is N for N. This seems to not leave any mechanism for representing the deletion of 10 NT (such as ACGGCGGCAA) at a position and the insertion of 3 NT (such as TTT) in the same place. If SUB used to be a code which captured indels and then it was dropped from the MAF file standard, then perhaps we should not allow it here if the reference allele length does not equal the variant allele length

Reply:
line 282 already does the check to determine whether the lengths of the reference and variant alleles match